sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala

```diff
@@ -63,25 +65,13 @@ case class ParquetScanBuilder(
       // The rebase mode doesn't matter here because the filters are used to determine
       // whether they is convertible.
       LegacyBehaviorPolicy.CORRECTED)
-    parquetFilters.convertibleFilters(this.filters).toArray
+    parquetFilters.convertibleFilters(translateDataFilter).toArray
```

Comment:
Shall we just use `pushedDataFilters` here? It's useless to include partition filters here.

Reply:
Parquet pushFilter is a little different from the rest. If you look at the original `pushFilters` here https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala#L74, it doesn't really do the push down logic here. That's why I didn't override `pushDataFilters` in parquet.  I can't use `pushedDataFilters` here because it's empty. Have to use `translateDataFilter()`

Reply:
Why do we care about partition filters in parquet filter pushdown? Can you look into the source code and find the reason? Otherwise, we should always use data filters. The previous code can be wrong/sub-optimal

Reply:
I think here is the explanation why  the actual parquet filter push down happens in `ParquetPartitionReaderFactory`.
https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala#L78

Reply:
Actually it doesn't contain the partition filters. In `translateDataFilter()`, I only have the data filters
```
for (filterExpr <- dataFilters) {
        val translated = DataSourceStrategy.translateFilter(filterExpr, true)
        . . .
```

Reply:
then can we implement `pushDataFilters` in parquet and mark all data filters as pushed? then here we can use `pushedDataFilters`